javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/ConfigBuilder.java

```diff
@@ -0,0 +1,182 @@
+import static io.opentelemetry.instrumentation.api.config.Config.DEFAULT_INTEGRATIONS_ENABLED;
+import static io.opentelemetry.instrumentation.api.config.Config.DEFAULT_KAFKA_CLIENT_PROPAGATION_ENABLED;
+import static io.opentelemetry.instrumentation.api.config.Config.DEFAULT_RUNTIME_CONTEXT_FIELD_INJECTION;
+import static io.opentelemetry.instrumentation.api.config.Config.DEFAULT_SQL_NORMALIZER_ENABLED;
+import static io.opentelemetry.instrumentation.api.config.Config.DEFAULT_TRACE_ENABLED;
+import static io.opentelemetry.instrumentation.api.config.Config.DEFAULT_TRACE_EXECUTORS_ALL;
+import static io.opentelemetry.instrumentation.api.config.Config.normalizePropertyName;
+
+import io.opentelemetry.instrumentation.api.config.Config;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.Function;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ConfigBuilder
+    extends io.opentelemetry.sdk.common.export.ConfigBuilder<ConfigBuilder> {
+  private static final Logger log = LoggerFactory.getLogger(ConfigBuilder.class);
+
+  public static final String EXPORTER_JAR = "otel.exporter.jar";
+  public static final String EXPORTER = "otel.exporter";
+  public static final String PROPAGATORS = "otel.propagators";
+  public static final String TRACE_ENABLED = "otel.trace.enabled";
+  public static final String INTEGRATIONS_ENABLED = "otel.integrations.enabled";
+  public static final String TRACE_CLASSES_EXCLUDE = "otel.trace.classes.exclude";
+  public static final String RUNTIME_CONTEXT_FIELD_INJECTION =
+      "otel.trace.runtime.context.field.injection";
+  public static final String TRACE_ANNOTATIONS = "otel.trace.annotations";
+  public static final String TRACE_METHODS = "otel.trace.methods";
+  public static final String TRACE_ANNOTATED_METHODS_EXCLUDE =
+      "otel.trace.annotated.methods.exclude";
+  public static final String TRACE_EXECUTORS_ALL = "otel.trace.executors.all";
+  public static final String TRACE_EXECUTORS = "otel.trace.executors";
+  public static final String SQL_NORMALIZER_ENABLED = "sql.normalizer.enabled";
+  public static final String KAFKA_CLIENT_PROPAGATION_ENABLED =
+      "otel.kafka.client.propagation.enabled";
+  public static final String HYSTRIX_TAGS_ENABLED = "otel.hystrix.tags.enabled";
+  public static final String ENDPOINT_PEER_SERVICE_MAPPING = "otel.endpoint.peer.service.mapping";
+
+  private final Map<String, String> allProperties = new HashMap<>();
+
+  @Override
+  public ConfigBuilder readProperties(Properties properties) {
+    return this.fromConfigMap(normalizedProperties(properties), NamingConvention.DOT);
```

Comment:
i think only integrationNames need this normalization, and can be done special for them in `isIntegrationEnabled`
```suggestion
    return this.fromConfigMap(properties, NamingConvention.DOT);
```